etc/dbus-serialbattery/bms/jkbms_brn.py

```diff
@@ -406,10 +414,34 @@ async def enable_charging(self, c):
         # data is 01 00 00 00 for on  00 00 00 00 for off;
         # the following bytes up to 19 are unclear and changing
         # dynamically -> auth-mechanism?
-        await self.write_register(0x1D, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x1E, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x1F, b"\x01\x00\x00\x00", 4, c)
-        await self.write_register(0x40, b"\x01\x00\x00\x00", 4, c)
+        await self.write_register(0x1D, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x1E, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x1F, b"\x01\x00\x00\x00", 4, c, True)
+        await self.write_register(0x40, b"\x01\x00\x00\x00", 4, c, True)
+
+    def jk_float_to_hex_little(self, val: float):
+        intval = int(val * 1000)
+        hexval = f'{intval:0>8X}'
+        return bytearray.fromhex(hexval)[::-1]
+
+    async def reset_soc_jk(self, c):
+        # Lowering OVPR / OVP
+        # That will trigger a High Voltage Alert
+        ovpr_trigger = round(self.bms_status["settings"]["cell_ovpr"] - 0.250, 3)
+        ovp_trigger = round(self.bms_status["settings"]["cell_ovp"] - 0.250, 3)
+        await self.write_register(JK_REGISTER_OVPR, self.jk_float_to_hex_little(ovpr_trigger), 0x04, c, True)
+        await self.write_register(JK_REGISTER_OVP, self.jk_float_to_hex_little(ovp_trigger), 0x04, c, True)
+
+        # Give BMS some time to recognize
+        await asyncio.sleep(5)
+
+        # Set values back to default
+        ovp_reset = round(self.bms_status["settings"]["cell_ovp"], 3)
+        ovpr_reset = round(self.bms_status["settings"]["cell_ovpr"], 3)
```

Comment:
This should be before all other, else if the data refreshes you have lost the old values.